Lib/shutil.py

```diff
@@ -446,43 +466,51 @@ def copytree(src, dst, symlinks=False, ignore=None, copy_function=copy2,
     function that supports the same signature (like copy()) can be used.
 
     """
-    names = os.listdir(src)
+    # TODO: use with statement; omitted just to make code review easier
+    # (will add later)
+    entries = os.scandir(src)
     if ignore is not None:
+        ls = []
+        names = set()
+        for entry in entries:
+            ls.append(entry)
```

Comment:
Why not just?
```python
entries = list(entries)
names = {entry.name for entry in entries}
```

Reply:
I did that in order to avoid iterating over the same sequence twice (1 for loop instead of 2).